test/1_stdlib/Runtime.swift

```diff
@@ -823,6 +823,48 @@ Runtime.test("casting AnyObject to class metatypes") {
   }
 }
 
+@inline(never)
+public func anyToAny<T, U>(a: T, _ : U.Type) -> U {
+  return a as! U
+}
+@inline(never)
+public func anyToAnyOrNil<T, U>(a: T, _ : U.Type) -> U? {
+  return a as? U
+}
+
+Runtime.test("Casting Optional") {
+  let x = SomeClass()
+  let sx : SomeClass? = x
+  let nx : SomeClass? = nil
+  print(anyToAny(x, Optional<SomeClass>.self)! === x)
+  print(anyToAny(sx, SomeClass.self) === x)
+  print(anyToAny(sx, Optional<SomeClass>.self)! === x)
+
+  print(anyToAny(nx, Optional<SomeClass>.self) == nil)
+  print(anyToAnyOrNil(nx, SomeClass.self) == nil)
+
+  let i = Int.max
+  let si : Int? = Int.max
+  let ni : Int? = nil
+  print(anyToAny(i, Optional<Int>.self)! == Int.max)
+  print(anyToAny(si, Int.self) == Int.max)
+  print(anyToAny(si, Optional<Int>.self)! == Int.max)
+
+  print(anyToAny(ni, Optional<Int>.self) == nil)
+  print(anyToAnyOrNil(ni, Int.self) == nil)
+
+  let ssx : SomeClass?? = sx
+  print(anyToAny(ssx, Optional<SomeClass>.self)! === x)
+  print(anyToAny(x, Optional<Optional<SomeClass>>.self)!! === x)
+  print(anyToAnyOrNil(ni, Int.self) == nil)
```

Comment:
Oops. I decided to move these all to an Interpreter test case, rewrote them all as prints, then changed my mind.